storage/storage_src.go

```diff
@@ -136,8 +137,10 @@ func (s *storageImageSource) GetBlob(ctx context.Context, info types.BlobInfo, c
 	}()
 	// On Unix and modern Windows (2022 at least) we can eagerly unlink the file to ensure it's automatically
 	// cleaned up on process termination (or if the caller forgets to invoke Close())
+	// On older versions of Windows we will have to fallback to relying on the caller to invoke Close()
+	removeFailure := false
```

Comment:
This reads to me as “remove a failure”.

Maybe `tmpFileRemovePending`, originally set to `true`, and cleared on success here?